Formula/aerc.rb

```diff
@@ -0,0 +1,34 @@
+class Aerc < Formula
+  include Language::Python::Virtualenv
+
+  desc "The world's best email client"
+  homepage "https://aerc-mail.org/"
+  url "https://git.sr.ht/~sircmpwn/aerc/archive/0.1.1.tar.gz"
+  sha256 "eb9f6cf6f77b1eef25d04d7c7cd05fdaebd3d41cd47fa95230b222acac94fc66"
+
+  depends_on "go" => :build
+  depends_on "scdoc" => :build
+
+  depends_on "dante"
+  depends_on "python"
+  depends_on "w3m"
+
+  resource "colorama" do
+    url "https://files.pythonhosted.org/packages/76/53/e785891dce0e2f2b9f4b4ff5bc6062a53332ed28833c7afede841f46a5db/colorama-0.4.1.tar.gz"
+    sha256 "05eed71e2e327246ad6b38c540c4a3117230b19679b875190486ddd2d721422d"
+  end
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+    venv.pip_install resources
+    inreplace "contrib/plaintext.py", "/usr/bin/env python3", "#{libexec}/bin/python3"
+    inreplace "contrib/hldiff.py", "/usr/bin/env python3", "#{libexec}/bin/python3"
+
+    system "make", "LDFLAGS=", "PREFIX=#{prefix}"
+    system "make", "install", "PREFIX=#{prefix}"
+  end
+  test do
+    # TODO: aerc doesn't support any flags. Add simple test when `aerc --version` is available
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Unfortunately` aerc` doesn't seem to have any command line only options yet, but it does log to stdout.

Would it be acceptable if the test used a tool like `timeout` to run the command with some invalid credentials and then checked the log to see if it tried to connect?

If so a solution could look something like this:
```bash
touch ~/.config/aerc/aerc.conf ~/.config/aerc/binds.conf
cat << CONF
[Test]
source   = imaps://test%40test.com@test.com
outgoing = smtps+plain://test%40test.com@test.com
default  = INBOX
from     = Test User <test@test.com>
CONF > ~/.config/aerc/accounts.conf
timeout 5s aerc > /tmp/aerc.log
# test that the log looks like expected
```

I'd imagine that the paths would have to be cleaned up to work with brew and I think `coreutils` might be a requirement for `timeout`, but the above works well enough on my machine.

Reply:
I absolutely understand that `--version` is a terrible test.  The lack of command-line options and complexity of a mail client made me realize there wasn't a quick win for writing a test. Even the above makes coreutils a build-time requirement which feels overkill.

Reply:
Is the test really essential? I mean it just sounds like a pain in the arse.

Reply:
@nnathan Yes, it is essential. It sounds like "a pain in the arse" to you because you don't have experience running a package manager.

Reply:
> Would it be acceptable if the test used a tool like `timeout` to run the command with some invalid credentials and then checked the log to see if it tried to connect?

Yes, this sounds like perhaps the best test we can do considering.

Reply:
There are some tests.

    go test ./...

Will be sufficient.

Reply:
No, it will not. These are runtime integration tests when the source code is no longer available. It is a test of our packaging and linking rather than unit tests distributed with the application.

Reply:
I see. Well, I'll add a `-v` flag momentarily.

Reply:
@ddevault If you're going to add something to the application specially it'd be great if you could make something a little more involved. Ideally something that e.g. spins up the main "GUI" and then immediately tears it down. Verifying that doesn't immediately crash would be an excellent test.

Reply:
A version flag is broadly useful and easy to add, but a "spin up the frontend" test is a lot less so in both respects. I've got other stuff to do but I can quickly get the version out of the way for you now and unblock your packaging work.

Reply:
@ddevault Sure, that's reasonable. A nice extension to `--version` would be the version string and any other e.g. compile-time conditional values such as the presence/use/versions of dante/python/w3m/scdoc/go. See `curl --version` for a nice example of this.

Reply:
aerc doesn't have any compile-time conditional variables outside of e.g. the installation prefix. Anyway, apply this patch to get `aerc -v`:

https://git.sr.ht/~sircmpwn/aerc/commit/74af57b6c617382ed0e8cd79626cac98ffc2e549.patch

Reply:
@ddevault If the installation prefix is hardcoded and/or used at all by the application: outputting it would be useful if that's at all possible?

Reply:
It's possible, and of some use, but I have to weigh the complexity add against the value add, and in this case I don't think it adds up.

Reply:
@ddevault Fair enough, thanks anyway.

Reply:
> @nnathan Yes, it is essential. It sounds like "a pain in the arse" to you because you don't have experience running a package manager.

Oi, I have written and run my own package manager, it's essentially a rite of passage for any Slackware user :).